graphql/client/client.go

```diff
@@ -42,11 +54,57 @@ func Services(cliCtx *cli.Context, duration schema.Duration) []schema.Service {
 	`)
 	request.Var("duration", duration)
 
-	ctx := context.Background()
-	if err := client.Run(ctx, request, &response); err != nil {
-		logger.Log.Fatalln(err)
-		panic(err)
+	executeQuery(cliCtx, request, &response)
+	return response["services"]
+}
+
+func Instances(cliCtx *cli.Context, nameOrID string, duration schema.Duration) []schema.ServiceInstance {
+	var serviceId string
+	service, err := searchServices(cliCtx, nameOrID, duration)
+	if err != nil {
+		serviceId = nameOrID
+	} else {
+		serviceId = service.ID
 	}
 
-	return response["services"]
+	var response map[string][]schema.ServiceInstance
+	request := graphql.NewRequest(`
+		query ($serviceId: ID!, $duration: Duration!) {
+    		instances: getServiceInstances(duration: $duration, serviceId: $serviceId) {
+      			id
+      			name
+      			language
+				instanceUUID
+      			attributes {
+        			name
+        			value
+      			}
+			}
+		}
+	`)
+	request.Var("serviceId", serviceId)
+	request.Var("duration", duration)
+
+	executeQuery(cliCtx, request, &response)
+	return response["instances"]
+}
+
+func searchServices(cliCtx *cli.Context, serviceName string, duration schema.Duration) (service schema.Service, err error) {
+	var response map[string][]schema.Service
+	request := graphql.NewRequest(`
+		query searchServices($keyword: String!, $duration: Duration!) {
+    		service: searchServices(duration: $duration, keyword: $keyword) {
```

Comment:
You should not use searchService. Use `searchService` by using service name as the parameter, this should be an equal match.

Reply:
done